Lib/test/datetimetester.py

```diff
@@ -2905,6 +2922,22 @@ def test_strftime(self):
         # A naive object replaces %z and %Z with empty strings.
         self.assertEqual(t.strftime("'%z' '%Z'"), "'' ''")
 
+    def test_strptime_invalid(self):
+        tests = [('2004-12-01 13:02:47.197',
+                  '%Y-%m-%d %H:%M:%S.%f'),
+                 ('2004-12-01', '%Y-%m-%d'),]
+        for date_string, date_format in tests:
```

Comment:
I think it would be good to use [`with self.subTest`](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) for these parametrized tests.

Also, per the other comment I guess you need to add something like `(`'12:30:15', '')` to get full coverage.

Reply:
I think you need two more test cases:

```python
    ('1900-01-01 12:30', '%Y-%m-%d %H:%M'),
    ('12:30:15', ''),
```

`date.strptime` has similarly missing tests.

Reply:
The fact that it doesn't raise an exception *is* an issue. I'm a bit surprised that it doesn't raise an exception on pure Python, that's a bug, because I'm pretty sure that:

`datetime.time.strptime("1901-01-01 12:30", "%Y-%m-%d %H:%M")` *does* raise an exception.